tools/scoring/score/dimensions/entity_connection_identification.py

```diff
@@ -27,41 +27,35 @@ class EntityConnectionIdentification(Dimension):
   Quantifies whether connections between entities were
   correctly and completely defined in the proposed file.
   """
-
-  # TODO: Figure out how to elegantly implement "facilities"
-  # and "equipment" categories given current object model
-  def __init__(self, *, deserialized_files: DeserializedFilesDict):
-    super().__init__(deserialized_files=deserialized_files)
-
-    proposed, solution = map(deserialized_files.get, (PROPOSED, SOLUTION))
-
-    # Isolate the connections from each dictionary of entities
-    solution_connections = [
-        tup for tup in (((cloud_device_id, connection)
-                         for connection in entity.connections)
-                        for cloud_device_id, entity in solution.items()
-                        if entity.connections is not None) for tup in tup
-    ]
-    proposed_connections = [
+  def _isolate_connections(self, file: DeserializedFile):
+    return [
         tup for tup in (((cloud_device_id, connection)
                          for connection in entity.connections)
-                        for cloud_device_id, entity in proposed.items()
+                        for cloud_device_id, entity in file.items()
                         if entity.connections is not None) for tup in tup
```

Comment:
there are 2x `for tup in tup`  is this legit?
can you please add docstring

Reply:
Yes, it is correct. (Nested list comprehensions are not the most intuitive.) 40f95b7179a6cd875f731b40685e6c9b24bfeb48

Reply:
Yes, it is correct. (Nested list comprehensions are not the most intuitive.) 40f95b7179a6cd875f731b40685e6c9b24bfeb48